### PR TITLE
Handle dataflow constraints in Alive backend

### PR DESCRIPTION
**Pull request by manasij7479**

### CODE REVIEW

File: lib/Infer/AliveDriver.cpp [hunk 1]
```diff
@@ -300,6 +301,26 @@ bool souper::AliveDriver::translateRoot(const souper::Inst *I, const Inst *PC,
   return true;
 }
 
+// Dummy because it doesn't actually build expressions.
+// It exists for the purpose of reusing parts of the abstract ExprBuilder here.
```
Comment (regehr) [inline lib/Infer/AliveDriver.cpp]: ugh... are these expected to be called? if not, make them throw a fatal error
also please put a FIXME to remove them when possible

File: lib/Infer/AliveDriver.cpp [hunk 2]
```diff
@@ -338,6 +359,9 @@ bool souper::AliveDriver::translateAndCache(const souper::Inst *I,
   switch (I->K) {
     case souper::Inst::Var: {
       ExprCache[I] = Builder.var(t, Name);
+      if (!translateDataflowFacts(I, F, ExprCache)) {
```
Comment (regehr) [inline lib/Infer/AliveDriver.cpp]: this should be
```return translateDataflowFacts(I, F, ExprCache);```

**Comment by regehr** [comment]
looks good other than minor nits, please fix and squash

**Comment by manasij7479** [comment]
Fixed

**Comment by manasij7479** [comment]
Bad news: Not all the demanded-bits tests pass with this patch.
Good news: Some do..

File: lib/Infer/AliveDriver.cpp [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #include "souper/Infer/AliveDriver.h"
+#include "souper/Extractor/ExprBuilder.h"
```
Comment (regehr) [inline lib/Infer/AliveDriver.cpp]: sorted, please

Comment (regehr) [reply lib/Infer/AliveDriver.cpp]: ping

**Comment by regehr** [comment]
LGTM, thanks!

**Pull request merged**